src/Analyzers/CSharp/CodeFixes/UseImplicitOrExplicitType/UseExplicitTypeCodeFixProvider.cs

```diff
@@ -74,42 +80,68 @@ public override Task RegisterCodeFixesAsync(CodeFixContext context)
             if (declarationContext is VariableDeclarationSyntax varDecl)
             {
                 typeSyntax = varDecl.Type;
+                variableIdentifier = varDecl.Variables.First().Identifier;
```

Comment:
Note in all branches here you're going from a syntax to the token, but in the use later on you are always doing token.Parent. Looks like you can switch variableIdentifier to just hold onto the parent syntax node.